File: CS 201 (Algorithm)/Lecture Qns/Python/Interval Containment.py
class Solution():
    def isContained(self,intervals: list[(int,int)]) -> bool:
        if not intervals:
            return False

        intervals.sort(key = lambda x: x[0])
        max_end_pt = intervals[0][1]


        for i in range(1,len(intervals)):
            if intervals[i][1] <= max_end_pt:
                return True

            max_end_pt = intervals[i][1]

        return False

sol = Solution()
print(sol.isContained([(1,5),(7,12),(2,10),(4,15)]))

File: CS 201 (Algorithm)/Lecture Qns/c++/Interval_Containment.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
    public:
        bool Int_Cont(vector<pair<int,int>> &intervals)
        {
            if (!intervals.size())
                return false;

            sort(intervals.begin(),intervals.end(),[](pair<int,int> a,pair<int,int>b)
            {
                return a.first <= b.first;
            });
            int i;
            int max_end_pt = intervals[0].second;
            for(i=1;i<intervals.size();i++)
            {
                if (intervals[i].second <= max_end_pt)
                    return true;
                
                max_end_pt = intervals[i].second;
            }
            return false;
        }
};

int main()
{
    Solution sol;
    vector<pair<int,int>> intervals = {{1,5},{2,10},{4,12},{7,15}};
    cout << sol.Int_Cont(intervals) << endl;
    return 0;
}